content/200-concepts/100-components/02-prisma-client/260-client-extensions/index.mdx

```diff
@@ -0,0 +1,159 @@
+- `result` extensions (for future release): [add custom results objects to Prisma Client](/concepts/components/prisma-client/client-extensions/result)
+- `query` extensions (for future release): [create custom Prisma Client queries](/concepts/components/prisma-client/client-extensions/query)
+
+</TopBlock>
+
+## About Prisma Client extensions
+
+To use a Prisma Client extension, you must create an associated _extended client_. An extended client is a forked variant of the standard Prisma Client with an extension applied. The standard client does not change.
+
+You can use [multiple independent extensions at the same time](#multiple-extensions). You can run an extension in its own isolated extended client, or combine it with other extensions in an extended client.
+
+You can also [share your Prisma Client extensions](/concepts/components/prisma-client/client-extensions/shared-extensions) with other Prisma users, and [import Prisma Client extensions developed by other users](#install-a-packaged-extension) into your Prisma project.
+
+### Client instances
+
+Extended clients interact with each other, and with the standard client, as follows:
+
+- Each extended client operates independently in an isolated instance.
+- Extended clients cannot conflict with each other, or with the standard client.
+- All extended clients and the standard client communicate with the same [Prisma query engine](/concepts/components/prisma-engines/query-engine).
+- All extended clients and the standard client share the same connection pool.
+
+#### Examples of isolated client instances
+
+Example uses of isolated client instances include the following:
+
+- Implement row-level security (RLS), where each http request has its own client with its own RLS extension, customized with session data. This can keep each user entirely separate, each in a separate client.
+- Add a `user.current()` method for the `user` model to get the current logged-in user.
+- Enable more verbose logging for requests if a debug cookie is set.
+- Attach a unique request id to all logs so you can correlate them later.
+- Remove a `delete` method from models unless the application calls the admin endpoint and the user has the necessary privileges.
+
+## Prerequisites
+
+To use Prisma Client extensions, you must enable the `clientExtensions` preview feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Create an extension
+
+Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an extended client. In your `$extends` structure, use the appropriate extension type or types (`model`, `client`, `result` or `query`). Write your extension logic in the `model`, `client`, `result` or `query` structure.
```

Comment:
Might it not be better to link to the reference of `$extends` directly instead of the section about client-level methods?

```suggestion
Use the [`$extends`](/reference/api-reference/prisma-client-reference#extends) client-level method to create an extended client. In your `$extends` structure, use the appropriate extension type or types (`model`, `client`, `result` or `query`). Write your extension logic in the `model`, `client`, `result` or `query` structure.
```

Reply:
Also, if we adopt the terminology of extension layers, this is probably another place where we should use it:

```suggestion
Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an extended client. In your `$extends` structure, use the appropriate extension layer (`model`, `client`, `result` or `query`). Write your extension logic in the `model`, `client`, `result` or `query` structure.
```

Reply:
I'll close this comment, pending a decision on whether to change the term that we use for the extension layers/types/TBD.